### PR TITLE
GG-18542: SQL: Introduce quotas for GroupBy aggregates.

### PR DESCRIPTION
**Pull request by AMashenkov**

### CODE REVIEW

File: modules/h2/src/main/java/org/h2/command/dml/SelectGroups.java [hunk 1]
```diff
@@ -119,13 +124,16 @@ public void done() {
...
             if (cursor.hasNext()) {
-                Map.Entry<ValueRow, Object[]> entry = cursor.next();
-                currentGroupByExprData = entry.getValue();
+                curEntry = cursor.next();
+                currentGroupByExprData = curEntry.getValue();
                 currentGroupRowId++;
-                return entry.getKey();
+                return curEntry.getKey();
             }
             return null;
```
Comment (ygerzhedovich) [inline modules/h2/src/main/java/org/h2/command/dml/SelectGroups.java]: seems need to set ``curEntry`` as ``null``

File: modules/h2/src/main/java/org/h2/command/dml/SelectGroups.java [hunk 2]
```diff
@@ -135,6 +143,8 @@ public void remove() {
             cursor.remove();
             currentGroupByExprData = null;
             currentGroupRowId--;
+
+            onUpdate(curEntry.getKey(), curEntry.getValue(), null);
```
Comment (ygerzhedovich) [inline modules/h2/src/main/java/org/h2/command/dml/SelectGroups.java]: is it possible NPE here?

File: modules/h2/src/main/java/org/h2/command/dml/SelectGroups.java [hunk 3]
```diff
@@ -439,4 +469,46 @@ public void nextLazyRow() {
...
+            return;
+
+        long size;
+
+        /* TODO: GG-18628: Estimate Aggregate objects size. */
+        if (row != null && old != null)
+            size = (row.length - old.length) * Constants.MEMORY_OBJECT;
+        else if (old == null) {
+            size = groupKey != null && groupKey.track() ? groupKey.getMemory() : 0;
+            size += Constants.MEMORY_ARRAY + row.length * Constants.MEMORY_OBJECT;
```
Comment (ygerzhedovich) [inline modules/h2/src/main/java/org/h2/command/dml/SelectGroups.java]: can be ``row`` ``null`` here?

File: modules/h2/src/main/java/org/h2/command/dml/SelectGroups.java (hunk 2 above)
Comment (AMashenkov) [reply modules/h2/src/main/java/org/h2/command/dml/SelectGroups.java]: NPE is impossible here as next() method is always called before this.
Moreover, Cursor.remove will fails with IllegalStateException before NPE happens.

File: modules/h2/src/main/java/org/h2/command/dml/SelectGroups.java (hunk 3 above)
Comment (AMashenkov) [reply modules/h2/src/main/java/org/h2/command/dml/SelectGroups.java]: No. I've add an assertion.

**Pull request closed without merging**